subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ResolutionStrategyFactory.java

```diff
@@ -0,0 +1,110 @@
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.internal.Factory;
+import org.gradle.internal.reflect.Instantiator;
+import org.gradle.internal.typeconversion.NotationParser;
+import org.gradle.vcs.internal.VcsMappingsStore;
+
+/**
+ * Creates fully initialized {@link ResolutionStrategyInternal} instances.
+ */
+public class ResolutionStrategyFactory implements Factory<ResolutionStrategyInternal> {
+
+    private final Instantiator instantiator;
+    private final DependencySubstitutionRules globalDependencySubstitutionRules;
+    private final VcsMappingsStore vcsMappingsStore;
+    private final ComponentIdentifierFactory componentIdentifierFactory;
+    private final ImmutableAttributesFactory attributesFactory;
+    private final ImmutableModuleIdentifierFactory moduleIdentifierFactory;
+    private final ComponentSelectorConverter componentSelectorConverter;
+    private final DependencyLockingProvider dependencyLockingProvider;
+    private final NotationParser<Object, ComponentSelector> moduleSelectorNotationParser;
+    private final ObjectFactory objectFactory;
+    private final StartParameterResolutionOverride startParameterResolutionOverride;
+    private final NotationParser<Object, Capability> capabilityNotationParser;
+
+    public ResolutionStrategyFactory(
+        Instantiator instantiator,
+        DependencySubstitutionRules globalDependencySubstitutionRules,
+        VcsMappingsStore vcsMappingsStore,
+        ComponentIdentifierFactory componentIdentifierFactory,
+        ImmutableAttributesFactory attributesFactory,
+        ImmutableModuleIdentifierFactory moduleIdentifierFactory,
+        ComponentSelectorConverter componentSelectorConverter,
+        DependencyLockingProvider dependencyLockingProvider,
+        NotationParser<Object, ComponentSelector> moduleSelectorNotationParser,
+        ObjectFactory objectFactory,
+        StartParameterResolutionOverride startParameterResolutionOverride
+    ) {
+        this.instantiator = instantiator;
+        this.globalDependencySubstitutionRules = globalDependencySubstitutionRules;
+        this.vcsMappingsStore = vcsMappingsStore;
+        this.componentIdentifierFactory = componentIdentifierFactory;
+        this.attributesFactory = attributesFactory;
+        this.moduleIdentifierFactory = moduleIdentifierFactory;
+        this.componentSelectorConverter = componentSelectorConverter;
+        this.dependencyLockingProvider = dependencyLockingProvider;
+        this.moduleSelectorNotationParser = moduleSelectorNotationParser;
+        this.objectFactory = objectFactory;
+        this.startParameterResolutionOverride = startParameterResolutionOverride;
+        this.capabilityNotationParser = new CapabilityNotationParserFactory(false).create();
+    }
+
+    @Override
+    public ResolutionStrategyInternal create() {
+        CapabilitiesResolutionInternal capabilitiesResolutionInternal = instantiator.newInstance(DefaultCapabilitiesResolution.class,
```

Comment:
💭 Should this be created in the constructor and cached to make repeated `ResolutionStrategy` creation slightly faster?

Reply:
This type is mutable and configurable by the user. We need a unique instance for each resolution strategy.